TensorMath.lua

```diff
@@ -257,6 +211,7 @@ for _,Tensor in ipairs({"ByteTensor", "CharTensor",
              end
    end
 
+   if Tensor ~= 'HalfTensor' then
```

Comment:
you are right, we don't have access to torch functions at this point.  There may be some clever things you can do with interface:print cwrap to properly guard everything with #ifdefs, or presumably building the libraries in multiple stages, so you have access to the functions you need when this is actually running.

In any case, since TH_NATIVE_HALF is basically a non-functional placeholder at this point anyway, it doesn't seem worth it to solve this issue right now.